src/main/java/mate/academy/springboot/datajpa/mappers/ProductMapper.java

```diff
@@ -0,0 +1,34 @@
+package mate.academy.springboot.datajpa.mappers;
+
+import mate.academy.springboot.datajpa.dto.ProductRequestDto;
+import mate.academy.springboot.datajpa.dto.ProductResponseDto;
+import mate.academy.springboot.datajpa.model.Product;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class ProductMapper {
+    private final CategoryMapper categoryMapper;
+
+    @Autowired
+    public ProductMapper(CategoryMapper categoryMapper) {
+        this.categoryMapper = categoryMapper;
+    }
```

Comment:
it's not consistent, use constructor or @RequiredArgsConstructor, but not both in your program